src/Controller/ApiController.php

```diff
@@ -49,16 +49,16 @@ public function __construct(Scheduler $scheduler, LoggerInterface $logger)
     /**
      * @Route("/packages.json", name="packages", defaults={"_format" = "json"}, methods={"GET"})
      */
-    public function packagesAction(string $webDir)
+    public function packagesAction(string $webDir): Response
     {
         // fallback if any of the dumped files exist
         $rootJson = $webDir.'/packages_root.json';
-        if (file_exists($rootJson)) {
-            return new Response(file_get_contents($rootJson));
+        if (file_exists($rootJson) && $content = file_get_contents($rootJson)) {
+            return new Response($content);
```

Comment:
Do you really need to read the whole content of the file into memory?
I would assume symfony would provide a Response class meant/optimized for this use case…?